src/modules/ekf2/EKF2.cpp

```diff
@@ -1802,15 +1802,14 @@ void EKF2::PublishStatus(const hrt_abstime &timestamp)
 	status.control_mode_flags = _ekf.control_status().value;
 	status.filter_fault_flags = _ekf.fault_status().value;
 
-	uint16_t innov_check_flags_temp = 0;
-	_ekf.get_innovation_test_status(innov_check_flags_temp, status.mag_test_ratio,
-					status.vel_test_ratio, status.pos_test_ratio,
-					status.hgt_test_ratio, status.tas_test_ratio,
-					status.hagl_test_ratio, status.beta_test_ratio);
-
-	// Bit mismatch between ecl and Firmware, combine the 2 first bits to preserve msg definition
-	// TODO: legacy use only, those flags are also in estimator_status_flags
-	status.innovation_check_flags = (innov_check_flags_temp >> 1) | (innov_check_flags_temp & 0x1);
+	status.mag_test_ratio = _ekf.getHeadingInnovationTestRatio();
+	status.vel_test_ratio = math::max(_ekf.getHorizontalVelocityInnovationTestRatio(),
```

Comment:
This might be problematic if there's no vertical velocity.

Reply:
But it's basically the same as it used to be, just filtered?

Reply:
Now if a vertical velocity isn't available getVerticalVelocityInnovationTestRatio() will return INFINITY and that will be the max for status.vel_test_ratio.

Reply:
is there a reason why you split horizontal and vertical up into 2 functions?

Reply:
In anticipation of replacing the preflight checker with the filtered test ratios as well. :smile: 

![image](https://github.com/PX4/PX4-Autopilot/assets/84712/6f261899-d02c-4900-8734-b6db420357a9)

(threshold TBD)

Reply:
So you want to keep it like that or handle the case?